pkg/operator/ceph/object/user/controller.go

```diff
@@ -296,6 +296,20 @@ func (r *ReconcileObjectStoreUser) createOrUpdateCephUser(u *cephv1.CephObjectSt
 	user, err = r.objContext.AdminOpsClient.GetUser(r.opManagerContext, *r.userConfig)
 	if err != nil {
 		if errors.Is(err, admin.ErrNoSuchUser) {
+			namspacedName := types.NamespacedName{Namespace: u.Namespace, Name: "rook-ceph-object-user-" + u.Spec.Store + u.Name}
+			cephObjectStoreUserSecret := &corev1.Secret{}
+			err := r.client.Get(r.clusterInfo.Context, namspacedName, cephObjectStoreUserSecret)
+			if err != nil {
+				if kerrors.IsNotFound(err) {
+					logger.Debugf("no user secret %q provided for cephobjectuser, using random values", "rook-ceph-object-user-"+u.Spec.Store+u.Name)
+				} else {
+					return errors.Wrapf(err, "failed to get user cephobjectuser secret %q", "rook-ceph-object-user-"+u.Spec.Store+u.Name)
+				}
+			} else {
+				r.userConfig.Keys[0].AccessKey = string(cephObjectStoreUserSecret.Data["AccessKey"])
```

Comment:
these are user defined values do we need to take care of there key rotation? As secrets might get obsolete?

Reply:
plus do we need to support encryption

Reply:
So this secret is created by the user before hand, and rook is not responsible for creating it. Is that's true, then rook should not be bother with rotating the keys.  User should be responsible to updating the keys in the secret and rook should be reconciling if the secret is updated.

Reply:
didn't understand what you meant by encryption here.